django/thunderstore/community/api/experimental/tests/test_api_package_listing.py

```diff
@@ -0,0 +1,71 @@
+import json
+
+import pytest
+from rest_framework.test import APIClient
+
+from conftest import TestUserTypes
+from thunderstore.community.models import PackageCategory, PackageListing
+from thunderstore.repository.models import TeamMember
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("user_type", TestUserTypes.options())
+def test_api_experimental_package_listing_update_user_types(
+    api_client: APIClient,
+    user_type: str,
+    active_package_listing: PackageListing,
+):
+    user = TestUserTypes.get_user_by_type(user_type)
+    if user not in TestUserTypes.fake_users():
+        api_client.force_authenticate(user=user)
+
+    response = api_client.post(
+        f"/api/experimental/package-listing/{active_package_listing.pk}/update/",
+        data=json.dumps({"categories": []}),
+        content_type="application/json",
+    )
+
+    expected_results = {
+        TestUserTypes.no_user: False,
+        TestUserTypes.unauthenticated: False,
+        TestUserTypes.regular_user: False,
+        TestUserTypes.deactivated_user: False,
+        TestUserTypes.service_account: False,
+        TestUserTypes.site_admin: True,
+        TestUserTypes.superuser: True,
+    }
+    should_succeed = expected_results[user_type]
+
+    print(response.content)
```

Comment:
Remove?

Reply:
Was intentionally left in since pytest only prints output for failing tests, and this was particularly useful when I had this test fail